src/analysis/exploration/searchModel.m

```diff
@@ -0,0 +1,155 @@
+            isAnnotation = true;
+            
+        elseif any(strcmp(cModelField,knownFields))
+            %If its a known field, than it is a cell array of strings and
+            %we will use it accordingly
+            isAnnotation = false;
+            if strcmp(cModelField,'mets')
+                if isempty(regexp(searchTerm,'\[[^\[]\]$'))
+                    fieldToUse = regexprep(model.mets,'\[[^\[]\]$','');
+                else
+                    fieldToUse = model.mets;
+                end
+            else
+                fieldToUse = model.(cModelField);
+            end
+        else      
+            if any(strcmp(cModelField,annotationsFields))
+                %If its an annotation field, we look into it.
+                fieldToUse = model.(cModelField);
+                isAnnotation = true;
+            else
+                %Field does nto match anything. don't search in it.
+                continue 
+            end
+        end        
+        %Find matches
+        [matchingIDs,positions,csims] = findMatchingFieldEntries(fieldToUse,searchTerm,isAnnotation,similarity);
+        if ~isempty(positions)
+            resultList(positions,modelField) = matchingIDs;            
+            similarities(positions) = max(similarities(positions),csims);
+        end
+    end
+    if any(any(~cellfun(@isempty, resultList)))
+        if printLevel > 0 && displayedFields > 0
+            fprintf('\n-----------------------------------------------------\n\n');
+        end
+        displayedFields = displayedFields + 1;
+        if printLevel > 0            
+            fprintf('The following %s have matching properties:\n\n',baseFieldNames{field});            
+        end
+        %get the base field
+        results.(cField) = struct();
+        relRows = ~all(cellfun(@isempty,resultList),2);
+        results.(cField).id = 'start';        
+        results.(cField).matches = struct();
+        results.(cField)(sum(relRows)).id = 'end';
+        %And the relevant results for that base field
+        relResults = resultList(relRows,:);
+        relResultPos = find(relRows);
+        %ORder according to highest similarity
+        [~,simorder] = sort(similarities(relRows),'descend');
+        for cResults = 1:size(relResults,1)
+            results.(cField)(cResults).id = model.(cField){relResultPos(simorder(cResults))};            
+            results.(cField)(cResults).matches = struct();
+            %Init struct with size
+            results.(cField)(cResults).matches.source = '';
+            results.(cField)(cResults).matches.value = '';
+            %get the field which have matching entries.
+            resultEntries = find(~cellfun(@isempty, relResults(simorder(cResults),:)));
+            results.(cField)(cResults).matches(numel(resultEntries)).source = '';
+            for cResult = 1:numel(resultEntries)
+                %Set the source to the fieldName and the value to the found
```

Comment:
May you only leave the comments on one line?

Reply:
I blame Matlabs autowrap...